torchmetrics/text/sacre_bleu.py

```diff
@@ -118,10 +118,10 @@ def update(  # type: ignore
             reference_corpus: An iterable of iterables of reference corpus
             translate_corpus: An iterable of machine translated corpus
         """
-        reference_corpus_: Sequence[Sequence[Sequence[str]]] = [
-            [self.tokenizer(line) for line in reference] for reference in reference_corpus
+        reference_corpus_: Sequence[Sequence[str]] = [
+            [" ".join(self.tokenizer(line)) for line in reference] for reference in reference_corpus
```

Comment:
I did `join` in sacre_bleu so it can fun `_bleu_score_update`. I kept it in sacre_bleu class, since I thought it'd be weird to do join in a tokenizer. Not sure if the maintainers agree with me here.

Reply:
Same thing with : https://github.com/PyTorchLightning/metrics/pull/640/files#diff-27e19aa3339eb07c921ebc841c0522aa4ab4b23ea827218ac19c52395613f7b2R338

Reply:
Actually, I would prefer for `SacreBLEU` to move tokenization into the `update` step as well instead.

Reply:
To clarify, do you mean you prefer to move tokenization into `bleu_score_update` -- as its already in `update`?

Reply:
do u mean to move it to bleu_score_update -- as it's already in `update`?

Reply:
I would suppose the following solution if @Borda and @SkafteNicki will agree. I'd add a new argument `tokenizer` into the `_bleu_score_update` method s.t. 
```
tokenizer: Union[Callable[str], Sequence[str]], _SacreBLEUTokenizer]
```

A default method for the base `BLEUScore` would be
```
def tokenize_fn(sentence: str) -> Sequence[str]:
    return sentence.split()
```
and for `SacreBLEUScore` an instance of `_SacreBLEUTokenizer` (similar to the module metric) would be used.

This solution would ensure we will not duplicate code for tokenization in functional and module metrics either for `BLEUScore` or `SacreBLEUScore`

Reply:
Waiting until @Borda and @SkafteNicki  to respond before implementation.

Reply:
I am fine with the changes proposed by @stancld. If we have chance to cut down on code duplicates then it would be great :]

Reply:
k will implement. Thank you for the suggestion @SkafteNicki  and @stancld !